app/replica.go

```diff
@@ -246,55 +309,32 @@ func startReplica(c *cli.Context) error {
 			syncResp <- cmd.Run()
 		}()
 	}
-	if frontendIP != "" {
-		if address == ":9502" {
-			address = "localhost:9502"
-		}
-		go AutoConfigureReplica(s, frontendIP, "tcp://"+address, replicaType)
-	}
+	return nil
+}
 
-	for s.Replica() == nil {
-		logrus.Infof("Waiting for s.Replica() to be non nil")
-		time.Sleep(2 * time.Second)
-	}
-	if replicaType == "clone" && snapName != "" {
-		logrus.Infof("Starting clone process\n")
-		status := s.Replica().GetCloneStatus()
-		if status != "completed" {
-			logrus.Infof("Set clone status as inProgress")
-			if err = s.Replica().SetCloneStatus("inProgress"); err != nil {
-				logrus.Error("Error in setting the clone status as 'inProgress'")
-				return err
-			}
-			if err = CloneReplica(s, "tcp://"+address, cloneIP, snapName); err != nil {
-				logrus.Error("Error in cloning replica, setting clone status as 'error'")
-				if statusErr := s.Replica().SetCloneStatus("error"); err != nil {
-					logrus.Errorf("Error in setting the clone status as 'error', found error:%v", statusErr)
-					return err
-				}
-				return err
-			}
-		}
-		logrus.Infof("Set clone status as Completed")
-		if err := s.Replica().SetCloneStatus("completed"); err != nil {
-			logrus.Error("Error in setting the clone status as 'completed'")
+func startClone(s *replica.Server, snapName, address, cloneIP string) error {
```

Comment:
good one

Reply:
can we pass the arguments as a structure `cloneConfig`

Reply:
Instead of creating a new structure, can we utilize the structure(_to be created, not part of this code yet_) returned by the function `fetchValues`/`getStartConfig`? This returned structure already has the arguments being passed in the function `startClone` and a few more arguments.